app/controllers/api/v1/authentication_controller.rb

```diff
@@ -186,14 +189,6 @@ def authenticated_user
         warden.authenticated? && (is_browse_app? || @user.allowed_login?(app_name))
       end
 
-      def current_user_channels
-        Channel.channels_for_user_with_app_context(current_user, app_name)
-      end
-
-      def browse_channels
-        current_user_channels << "browse"
-      end
```

Comment:
@abhinav098 a brief note here about this change. You've correctly identified the problem - that we need to ensure the browse app is subscribing to the `browse` channel. However, I've removed the `browse_channels` method because it contains implementation detail about that specific channel. Instead, that should go inside the `Channel` class and not in a controller. (Principle of Separation of Concerns.) A controller cares about receiving a request and sending a response but it shouldn't need to know anything about channels or the details of the data it is sending.

Here we move the logic that decides whether to include the `browse` channel into the Channel.rb alongside the other logic that decides when to include the `reviewer` channel or the `order_fulfilment` channel. This groups the logic together and cleans up the code.

I'm sure all of this is already clear but felt it worth highlighting in more detail as we can apply this principle in many places to our API code.

Reply:
yeah, that makes sense @steveyken . Thanks